metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -2019,6 +2020,29 @@ def strip_time_from_dt(ts: dt.datetime) -> dt.datetime:
         )
 
 
+@dataclass(frozen=True)
+class CteGenerationResult:
+    """This stores parameters for creating a dataset from a CTE."""
+
+    source_dataflow_plan_node_id: NodeId
+    cte_node: SqlCteNode
+    select_columns: Tuple[SqlSelectColumn, ...]
+    instance_set: InstanceSet
+
+    def get_sql_data_set(self) -> SqlDataSet:
+        """Return a dataset that represents reading from the CTE."""
+        cte_alias = self.cte_node.cte_alias
+        return SqlDataSet(
+            instance_set=self.instance_set,
+            sql_select_node=SqlSelectStatementNode.create(
```

Comment:
The change I mentioned above would also allow you to access the `SqlSelectStatementNode` via `self.cte_node.select_statement` instead of building a new one here.